### PR TITLE
python-modernize . -w -f lib2to3.fixes.fix_itertools

### PR DESCRIPTION
**Pull request by nickpell**

### CODE REVIEW

**Comment by nickpell** [comment]
@dimagi/py3 cc: @orangejenny @calellowitz

File: corehq/apps/app_manager/suite_xml/sections/entries.py [hunk 1]
```diff
@@ -791,7 +792,7 @@ def append_update(dict_, new_dict):
             datum_ids = {d.datum.id: d for d in datums}
             index = 0
             changed_ids_by_case_tag = defaultdict(list)
-            for this_datum_meta, parent_datum_meta in list(izip_longest(datums, parent_datums)):
+            for this_datum_meta, parent_datum_meta in list(zip_longest(datums, parent_datums)):
```
Comment (esoergel) [inline corehq/apps/app_manager/suite_xml/sections/entries.py]: nit: this `list()` call is unnecessary in an iterative context

**Review by esoergel — APPROVED** [review]

File: corehq/apps/domain/models.py [hunk 2]
```diff
@@ -635,7 +635,7 @@ def get_all(cls, include_docs=True):
         if not include_docs:
             return domains
         else:
-            return imap(cls.wrap, iter_docs(cls.get_db(), [d['id'] for d in domains]))
+            return map(cls.wrap, iter_docs(cls.get_db(), [d['id'] for d in domains]))
```
Comment (emord) [inline corehq/apps/domain/models.py]: Will pylint/stickler catch future uses of map?

Comment (nickpell) [reply corehq/apps/domain/models.py]: I *think* it's smart enough to tell the difference between the builtin map and an imported map

File: corehq/apps/app_manager/suite_xml/sections/entries.py (hunk 1 above)
Comment (millerdev) [reply corehq/apps/app_manager/suite_xml/sections/entries.py]: I think it's necessary because of line 803: `datums.insert(...`

Comment (esoergel) [reply corehq/apps/app_manager/suite_xml/sections/entries.py]: 😱

**Pull request merged**